src/services/completions.ts

```diff
@@ -852,15 +858,16 @@ namespace ts.Completions {
         });
         const importAdder = codefix.createImportAdder(sourceFile, program, preferences, host);
 
+        // Create empty body for possible method implementation.
         let body;
         if (preferences.includeCompletionsWithSnippetText) {
             isSnippet = true;
             // We are adding a tabstop (i.e. `$0`) in the body of the suggested member,
             // if it has one, so that the cursor ends up in the body once the completion is inserted.
             // Note: this assumes we won't have more than one body in the completion nodes, which should be the case.
-            const emptyStatement = factory.createExpressionStatement(factory.createIdentifier(""));
-            setSnippetElement(emptyStatement, { kind: SnippetKind.TabStop, order: 0 });
-            body = factory.createBlock([emptyStatement], /* multiline */ true);
+            const emptyStmt = factory.createEmptyStatement();

```

Comment:
conceptually, it only makes sense for a tab stop snippet to be attached to a node that doesn't cause anything to be emitted. using an empty identifier, as I was before, was causing some troubles when setting positions for synthetic nodes, because we can't assume the identifier will be empty. an empty statement seemed more appropriate then.

Reply:
I can’t tell if you already got this, but an empty statement is _only_ parsable when terminated by a semicolon, so in the reverse, it’s impossible to emit an empty statement without emitting a semicolon. Is the formatter removing this semicolon or did you somehow get this to emit without one?

Reply:
I hadn't thought of that, I think. The semicolon doesn't exist in this case because the empty statement is never emitted -- when we call `emitTabStop`, it doesn't emit the node (empty statement) the tab stop is attached to. Do you think that could be a problem?

Reply:
Got it. I don’t think it’s a problem, it’s just a little weird to think about.

Reply:
it's definitely weird. I tried to go for the least weird option :/